### PR TITLE
ci: Only run CI tests when relevant files are changed

### PR DESCRIPTION
**Pull request by WillDaSilva**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,9 +1,21 @@
 name: Tests
 
 on:
-  pull_request: {}
+  pull_request:
+    paths:
+      - src/**
```
Comment (edgarrmondragon) [inline .github/workflows/test.yml:6]:
```suggestion
      - src/**
      - tests/**
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+      - noxfile.py
+      - pyproject.toml
+      - .github/workflows/test.yml
+      - .github/workflows/resources/constraints.txt
   push:
     branches: [main]
+    paths:
+      - src/**
```
Comment (edgarrmondragon) [inline .github/workflows/test.yml:14]:
```suggestion
      - src/**
      - tests/**
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (WillDaSilva) [reply .github/workflows/test.yml:6]: :facepalm:

File: .github/workflows/test.yml (hunk 2 above)
Comment (WillDaSilva) [reply .github/workflows/test.yml:14]: Thanks for catching this

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**